tf_encrypted/keras/engine/base_layer.py

```diff
@@ -135,6 +136,8 @@ def set_weights(self, weights, sess=None):
         sess.run(tfe.assign(w, tfe_weights_pl), feed_dict=fd)
     elif isinstance(weights[0], PondPrivateTensor):
       for i, w in enumerate(self.weights):
+        if not sess.run(KE.is_variable_initialized(weights[i])):
```

Comment:
might be worth changing the order of `if isinstance` and `for i, w`, should allow the user to provide a mix of different types